### PR TITLE
In load_vcf, when passed a URL download it first to a local file then…

### LINKED ISSUE #163: maximum recursion depth exceeded when loading a vcf from a URL
For both `load_vcf` and `load_vcf_fast` we seem to end up in an infinite recursion involving url parsing:

```
In [3]: varcode.__version__
Out[3]: '0.4.16+1.g84db1b3'

In [5]: x = varcode.load_vcf("https://github.com/hammerlab/varcode/blob/master/test/data/multiallelic.vcf")
...
/Users/tim/sinai/git/varcode/varcode/vcf.pyc in load_vcf(path, genome, reference_vcf_key, only_passing, allow_extended_nucleotides, include_info, chunk_size, max_variants)
     84
     85     require_string(path, "Path or URL to VCF")
---> 86     parsed_path = parse_url_or_path(path)
     87
     88     if parsed_path.scheme and parsed_path.scheme.lower() != "file":

/Users/tim/sinai/git/varcode/varcode/vcf.pyc in parse_url_or_path(s)
    458     if s.startswith("//"):
    459         s = "/" + s.lstrip("/")
--> 460     return urllib.parse.urlparse(s)

/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urlparse.pyc in urlparse(url, scheme, allow_fragments)
    141     Note that we don't break the components up in smaller bits
    142     (e.g. netloc is a single string) and we don't expand % escapes."""
--> 143     tuple = urlsplit(url, scheme, allow_fragments)
    144     scheme, netloc, url, query, fragment = tuple
    145     if scheme in uses_params and ';' in url:

/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urlparse.pyc in urlsplit(url, scheme, allow_fragments)
    174     allow_fragments = bool(allow_fragments)
    175     key = url, scheme, allow_fragments, type(url), type(scheme)
--> 176     cached = _parse_cache.get(key, None)
    177     if cached:
    178         return cached

RuntimeError: maximum recursion depth exceeded in cmp
```

Can also be reproduced by running:

```
$ RUN_TESTS_REQUIRING_INTERNET=1 nosetests test/test_vcf.py
```

### PR DESCRIPTION
**Pull request by timodonnell**
… open that.

Closes #163

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/varcode/165)

### CODE REVIEW

File: varcode/vcf.py [hunk 1]
```diff
@@ -91,14 +92,24 @@ def load_vcf(
         # implementation of read_table (with engine="python") helps with some
         # issues but introduces a new set of problems (e.g. the dtype parameter
         # is not accepted). For these reasons, we're currently not attempting
-        # to load VCFs over HTTP with pandas, and fall back to the pyvcf
-        # implementation here.
+        # to load VCFs over HTTP with pandas directly, and instead download it
+        # to a temporary file and open that.
+        (filename, headers) = urllib.request.urlretrieve(path)
```
Comment (iskandr) [inline varcode/vcf.py]: What happens if `load_vcf` fails? Where does newly downloaded file stick around?

**Comment by timodonnell** [comment]
Good catch! Updated it to cleanup the file.

**Comment by iskandr** [comment]
Good to merge?

**Comment by timodonnell** [comment]
Yep

**Pull request merged**